### PR TITLE
TDI-43491 : fix NPE on datetime

### PR DESCRIPTION
**Pull request by rlecomte-talend**

### CODE REVIEW

File: bigquery/src/main/java/org/talend/components/bigquery/output/TacoKitRecordToTableRowConverter.java [hunk 1]
```diff
@@ -132,7 +132,9 @@ public TacoKitRecordToTableRowConverter(com.google.cloud.bigquery.Schema tableSc
                     tableRow.put(fieldName, input.getBoolean(fieldName));
                     break;
                 case DATETIME:
-                    tableRow.put(fieldName, getDateFunction(field.getType()).apply(input.getDateTime(fieldName)));
+                    tableRow.put(fieldName, Optional.ofNullable(input.getDateTime(fieldName))
+                            .map( dt -> getDateFunction(field.getType()).apply(dt))
+                            .orElse(null));
```
Comment (undx) [inline bigquery/src/main/java/org/talend/components/bigquery/output/TacoKitRecordToTableRowConverter.java]: Do you think that we may have some nulls in array (L107) ?

File: bigquery/src/test/java/org/talend/components/bigquery/output/TacoKitRecordToTableRowConverterTest.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+import org.talend.sdk.component.runtime.record.RecordBuilderFactoryImpl;
+
+import java.util.Date;
+import java.util.Locale;
+import java.util.Map;
+
+public class TacoKitRecordToTableRowConverterTest {
+
+    @Test
+    public void testTDI_43491() {
```
Comment (undx) [inline bigquery/src/test/java/org/talend/components/bigquery/output/TacoKitRecordToTableRowConverterTest.java]: rename method to match test case.

File: bigquery/src/main/java/org/talend/components/bigquery/output/TacoKitRecordToTableRowConverter.java (hunk 1 above)
Comment (rlecomte-talend) [reply bigquery/src/main/java/org/talend/components/bigquery/output/TacoKitRecordToTableRowConverter.java]: I'll double check this one too

File: bigquery/src/main/java/org/talend/components/bigquery/output/TacoKitRecordToTableRowConverter.java [hunk 3]
```diff
@@ -24,10 +24,7 @@
 
 import java.time.ZonedDateTime;
 import java.time.format.DateTimeFormatter;
-import java.util.Collection;
-import java.util.HashMap;
-import java.util.Map;
-import java.util.Optional;
+import java.util.*;
```
Comment (undx) [inline bigquery/src/main/java/org/talend/components/bigquery/output/TacoKitRecordToTableRowConverter.java]: latest comment ;-) 
avoid star import

Comment (rlecomte-talend) [reply bigquery/src/main/java/org/talend/components/bigquery/output/TacoKitRecordToTableRowConverter.java]: Have to go deep inside Intellij settings to correct that

Comment (undx) [reply bigquery/src/main/java/org/talend/components/bigquery/output/TacoKitRecordToTableRowConverter.java]: sent idea config settings through slack

**Pull request merged**